src/core/exec-invoke.c

```diff
@@ -4873,6 +5001,32 @@ int exec_invoke(
                 }
         }
 
+        /* Unshare a new PID namespace before setting up mounts to ensure /proc is mounted with only processes in PID namespace visible.
+         * Note PrivatePIDs=yes implies MountAPIVFS=yes so we'll always ensure procfs is remounted. */
+        if (needs_sandboxing && exec_needs_pid_namespace(context) && params->pidref_transport_fd >= 0) {
+                r = can_mount_proc(context, params);
+                if (r > 0) {
+                        r = setup_private_pids(context, params);
+                        if (ERRNO_IS_NEG_PRIVILEGE(r))
+                                log_exec_warning_errno(context, params, r,
+                                                        "PrivatePIDs=yes is configured, but pid namespace setup failed due to lack of privileges, ignoring: %m");
+                        else if (r < 0) {
+                                *exit_status = EXIT_NAMESPACE;
+                                return log_exec_error_errno(context, params, r, "Failed to set up pid namespace: %m");
+                        }
+
+                        if (r > 0)
+                                /* Return value > 0 means we are in the parent process. Thus, return immediately in order to exit ASAP.
+                                * The rest of the setup will continue in the child process. */
+                                return 0;
+                } else if (r == 0)
+                        log_exec_warning(context, params, "PrivatePIDs=yes is configured, but /proc cannot be re-mounted due to lack of privileges, ignoring.");
```

Comment:
in log msgs and docs we usually want that path names that refer to dirs are suffixed with a slash. i.e. /proc → /proc/

Reply:
will fix docs too